### PR TITLE
15141 absolute URL

### PR DESCRIPTION
**Pull request by thomasmoors**
https://labela.atlassian.net/browse/BYNDR-15141

### CODE REVIEW

**Comment by joshp123** [comment]
@happylinks please ensure this is tested before merging, we reordered our procedure so it must be CRd before test instead of after.

File: includes/media_bynder.formatters.inc [hunk 1]
```diff
@@ -180,7 +180,102 @@ function media_bynder_preprocess_image(&$vars) {
...
+ * Returns the absolute URL for the custom calls
+ */
+function media_bynder_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+    if ($display['type'] == 'media_bynder_absolute_url') {
+        $settings = $display['settings'];
+        $effects = image_styles()[$settings['image_style']];
+
+        foreach ($effects as $effect) {
+            if (is_array($effect)) {
+                $effect = reset($effect);
```
Comment (happylinks) [inline includes/media_bynder.formatters.inc:260]: why use reset?

File: includes/media_bynder.formatters.inc [hunk 2]
```diff
@@ -180,7 +180,102 @@ function media_bynder_preprocess_image(&$vars) {
...
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ * Returns the absolute URL for the custom calls
+ */
+function media_bynder_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+    if ($display['type'] == 'media_bynder_absolute_url') {
+        $settings = $display['settings'];
+        $effects = image_styles()[$settings['image_style']];
```
Comment (happylinks) [inline includes/media_bynder.formatters.inc:255]: Is image_styles a global function?

File: includes/media_bynder.formatters.inc [hunk 3]
```diff
@@ -180,7 +180,102 @@ function media_bynder_preprocess_image(&$vars) {
...
+                    $effect_width = $effect['data']['width'];
+                    $effect_height = $effect['data']['height'];
+
+                    if (!empty($items)) {
+                        $media_item = array(
+                            'idHash' => media_bynder_idHash_from_uri($items[0]['uri'])
+                        );
+
+                        $element = array();
+                        $image_url = customResolutionUrl($media_item, $effect_width, $effect_height, $effect_name);
```
Comment (happylinks) [inline includes/media_bynder.formatters.inc:274]: same, global? please document.

File: includes/media_bynder.formatters.inc (hunk 2 above)
Comment (thomasmoors) [inline includes/media_bynder.formatters.inc:255]: It's a global Drupal function

File: includes/media_bynder.formatters.inc (hunk 1 above)
Comment (thomasmoors) [inline includes/media_bynder.formatters.inc:260]: To get the only element from an array from which the key is unknown

**Pull request merged**